cashctrl_ledger/ledger.py

```diff
@@ -21,13 +26,64 @@ class CashCtrlLedger(LedgerEngine):
 
     def __init__(self, client: Union[CachedCashCtrlClient, None] = None):
         super().__init__()
-        self._client = CachedCashCtrlClient() if client is None else client
+        client = CachedCashCtrlClient() if client is None else client
+        self._client = client
+        self._tax_codes = TaxCode(client=client, schema=TAX_CODE_SCHEMA)
+        self._accounts = Account(client=client, schema=ACCOUNT_SCHEMA)
 
     # ----------------------------------------------------------------------
     # File operations
 
+    def dump_to_zip(self, archive_path: str):
```

Comment:
Why do we need `dump_to_zip`, `restore_from_zip` and `restore`?
I thought theses were solely handled by the parent class.

Reply:
As noted in the PR description, our tests require a functioning dump/restore mechanism. However, since not all entities have been implemented yet, these methods will currently raise errors due to missing properties like `_ledger`, `assets`, and others.

To address this for now, I’ve implemented the necessary methods for the currently available entities only. This ensures the tests can run smoothly without interruptions while we continue to expand entity support.

Once all entities are implemented, we should remove any temporary or unnecessary parts to keep the codebase clean and maintainable.

Reply:
Thanks. Understood.

(This was not immediately clear to me after reading the PR description).